### PR TITLE
feat(Panda Novel): add presence

### LINKED ISSUE #7100: Panda Novel | www.panda-novel.com
### Website name

Panda Novel

### Website URL

https://www.panda-novel.com/

### Website logo

https://www.panda-novel.com/static/pc/images/logo.png

### Prerequisites

- [ ] It is a paid service
- [X] It displays NSFW content
- [ ] It is region restricted

### Description

show what novel you are reading from the site, novel name, chapter count /name, cover image

playing game: Panda Novel
[Picture of novel cover from site when on it, anywhere else panda novel cover]
Novel name
chapter 123 "wandering coder"
button to novel on site
button to chapter on site

just like this request https://github.com/PreMiD/Presences/issues/6006

Normal logo https://www.panda-novel.com/static/pc/images/logo.png
bilde

Night logo https://www.panda-novel.com/static/pc/images/logo-night.png 
bilde

### PR DESCRIPTION
**Pull request by shiseni**
## Description 
Add presence for [Panda Novel](https://www.panda-novel.com)
* Resolves #7100  

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

| Home | Novel Page | Author Page | Searching |
| --- | --- | --- | --- |
| image | image | image | image

</details>

### CODE REVIEW

File: websites/P/Panda Novel/presence.ts [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+			presence.getSetting<boolean>("logo"),
+			presence.getSetting<boolean>("time"),
+			presence.getSetting<boolean>("buttons"),
+		]),
+		{ pathname, href } = document.location,
+		path = pathname.split("/");
+
+	switch (path[1]) {
+		case "top":
+			presenceData.details = "Home";
```
Comment (DarkVillager) [inline websites/P/Panda Novel/presence.ts:33]: there's a string for that

File: websites/P/Panda Novel/presence.ts [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+
+		case "rankings":
+			presenceData.details = "Viewing reading top";
+			presenceData.state = textContent(".ranking-tabs .current");
+			presenceData.smallImageKey = Assets.Viewing;
+			presenceData.smallImageText = "Viewing";
+			break;
+
+		case "author":
+			presenceData.details = "Searching author";
```
Comment (DarkVillager) [inline websites/P/Panda Novel/presence.ts:83]:
```suggestion
			presenceData.details = "Searching authors for";
```

File: websites/P/Panda Novel/presence.ts [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+
+		case "browsenovel":
+			presenceData.details = "Searching novel";
+			presenceData.state = textContent(".filter-list .current");
+			presenceData.smallImageKey = Assets.Searching;
+			presenceData.smallImageText = "Searching";
+			break;
+
+		case "rankings":
+			presenceData.details = "Viewing reading top";
```
Comment (DarkVillager) [inline websites/P/Panda Novel/presence.ts:76]:
```suggestion
			presenceData.details = "Viewing top novels read";
```

File: websites/P/Panda Novel/presence.ts [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+
+		case "author":
+			presenceData.details = "Searching author";
+			presenceData.state = textContent(".ranking-tabs .current");
+			presenceData.smallImageKey = Assets.Searching;
+			presenceData.smallImageText = "Searching";
+			if (path[2] === "works") {
+				presenceData.details = "Viewing works of the author";
+				presenceData.state = textContent(".author-desc dl dd h4");
+				if (logo && !privacy) {
```
Comment (DarkVillager) [inline websites/P/Panda Novel/presence.ts:90]: why don't you just do this at the bottom?
```ts
	if (!logo && presenceData.largeImageKey !== Assets.Logo)
		presenceData.largeImageKey = Assets.Logo;
		```

File: websites/P/Panda Novel/presence.ts [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+								".breadcrumb li:nth-child(2) a"
+							)?.href,
+						},
+					];
+					break;
+			}
+			break;
+
+		case "newnovel":
+			presenceData.details = "Viewing week new novels";
```
Comment (DarkVillager) [inline websites/P/Panda Novel/presence.ts:151]:
```suggestion
			presenceData.details = "Viewing this weeks new novels";
```

File: websites/P/Panda Novel/presence.ts [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+			presenceData.details = "Searching novel";
+			presenceData.state =
+				document.querySelector<HTMLInputElement>("input")?.value;
+			presenceData.smallImageKey = Assets.Searching;
+			presenceData.smallImageText = "Searching";
+			if (path[2]) presenceData.state = textContent(".header-content h4 i");
+			break;
+
+		case "ucenter":
+			presenceData.details = "Viewing profile";
```
Comment (DarkVillager) [inline websites/P/Panda Novel/presence.ts:167]: https://github.com/PreMiD/Localization/blob/210cdd6e8d5268166f0a6e9e0a934246b9871baf/src/Presence/general.json#L306

**Review by DarkVillager — CHANGES_REQUESTED** [review]

File: websites/P/Panda Novel/presence.ts (hunk 4 above)
Comment (shiseni) [reply websites/P/Panda Novel/presence.ts]: > why don't you just do this at the bottom?

I do not know

**Review by EncryptedDev — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]

**Pull request merged**